core/state_processor_rip7560.go

```diff
@@ -399,45 +452,30 @@ func preparePostOpMessage(vpr *ValidationPhaseResult, chainConfig *params.ChainC
 }
 
 func validateAccountReturnData(data []byte) (uint64, uint64, error) {
-	MAGIC_VALUE_SENDER := uint32(0xbf45c166)
 	if len(data) != 32 {
 		return 0, 0, errors.New("invalid account return data length")
 	}
-	magicExpected := binary.BigEndian.Uint32(data[:4])
+	magicExpected, validUntil, validAfter := UnpackValidationData(data)
+	//todo: we check first 8 bytes of the 20-byte address (the rest is expected to be zeros)
 	if magicExpected != MAGIC_VALUE_SENDER {
+		if magicExpected == MAGIC_VALUE_SIGFAIL {
+			return 0, 0, errors.New("account signature error")
```

Comment:
You probably don't want to return it simply as an error. Error will stop the execution, meaning that i.e. paymaster frame will not run. AFAIK this is not what we want in the gas estimation.

Reply:
will fix when we add a test for estimateGas. I'm not sure what to return now.